core/src/pixelcolor/mod.rs

```diff
@@ -102,7 +103,13 @@ pub use web_colors::WebColors;
 /// See the [module-level documentation] for more details.
 ///
 /// [module-level documentation]: self
-pub trait PixelColor: Copy + PartialEq {
+pub trait PixelColor: Copy + PartialEq {}
+
+/// Storable pixel color.
+///
+/// A storable pixel color can be converted to and from a raw data type. All colors,
+/// that implement this trait, can be used with raw images and framebuffers.
+pub trait StorablePixelColor: PixelColor + From<Self::Raw> + Into<Self::Raw> {
```

Comment:
Most `PixelColor`s will have an in-memory representation, but I didn't want to make this mandatory. If someone wants to create a custom color type they might not be interested in defining an in-memory representation, e.g. if they use an enum and a match statement in their `DrawTarget` impl.

Reply:
That particular type already has `type Raw = ();`. I expect this to continue working, even if `()` doesn't implement certain traits that would allow me to pass the color to certain functions. I might be paper-engineering here but I don't expect this to be a hard problem.

Reply:
But `Raw = ()` isn't possible anymore, because `RawData` isn't implemented for `()` anymore. Having a dummy implementation of `RawData` for `()` always bugged me. This is much cleaner now IMO and `RawData` is now guaranteed to be storable/loadable on the type level, which helps to clean up the trait bounds in e.g. `ImageRaw`.

Reply:
> What would you expect a user to do if they aren't interested in an in-memory representation, but are required by the changed PixelColor trait to implement one?

Circling back to this question, I completely expect a display driver to support as wide a set of features as possible. If the authors choose not to, they can implement the dummy type themselves.

Reply:
> ... ,they can implement the dummy type themselves.

Which dummy type are you talking about? `RawData` is sealed and cannot be implemented externally.

Reply:
Eh fine, you've probably given this much more consideration than I have. I still think in the TriColor case the raw storage should just be defined, but 🤷‍♂️ In my opinion, users can get lazy and pick a random RawData type and mark `into_storage` `unimplemented!()` or something, but I'd definitely expect library developers to define a conversion. There will always be users who want to use "that one" feature. But again, this is my opinion, and I'll stop wasting your time on it :)